source/documentation/tools/create-a-derived-table/data-modelling-concepts.md

```diff
@@ -2,4 +2,388 @@
+
+
+## Fields, aggregations, and grouping
+
+- 🔙 Fields should be stated before aggregates and window functions.
+- 🤏🏻 Aggregations should be executed as early as possible (on the smallest data set possible) before joining to another table to improve performance.
+- 🔢 Grouping by a number (eg. group by 1, 2) is preferred over listing the column names (see [this classic rant](https://www.getdbt.com/blog/write-better-sql-a-defense-of-group-by-1) for why). Note that if you are grouping by more than a few columns, it may be worth revisiting your model design.
+- 🔢 Column names should be written out explicitly with the column names in order statements to avoid ambiguity. 
+Example:
+```SQL
+select
+    defendant_on_case_id,
+    result_priority,
+    custodial_period_days,
+    monetary,
+    final_offence_ho_code_priority,
+    final_offence_ho_code,
+    defendant_on_offence_id,
+    disposal2_id,
+    row_number()
+        over (
+            partition by
+                defendant_on_case_id
+            order by
+                result_priority asc nulls last,
+                custodial_period_days desc nulls last,
+                monetary desc nulls last,
+                final_offence_ho_code_priority asc nulls last,
+                final_offence_ho_code asc nulls last,
+                defendant_on_offence_id asc nulls last,
+                disposal2_id asc nulls last
+        ) as most_serious_disposal_rank,
+from disposals
+group by 1, 2, 3, 4, 5, 6, 7, 8
+order by defendant_on_case_id
+```
+
+## Joins
+
+- 👭🏻 Prefer `union all` to `union` unless you explicitly want to remove duplicates.
+- 👭🏻 If joining two or more tables, _always_ prefix your column names with the table name where that column is coming from. If only selecting from one table, prefixes are not needed.
+- 👭🏻 Be explicit about your join type (i.e. write `inner join` instead of `join`).
+- 🥸 Avoid table aliases in join conditions (especially initialisms) — it's harder to understand what the table called "c" is as compared to "customers".
+- ➡️ Always move left to right to make joins easy to reason about - `right joins` often indicate that you should change which table you select `from` and which one you `join` to.
+
+## 'Import' CTEs
+
+- 🔝 All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file.
+- 📦 'Import' CTEs should be named after the table they are referencing.
+- 'Import' CTEs should be short and concise to make it easy to read what tables are being read in. Try not to have lots of fields in your select statement as it makes reading the CTEs harder, `select *` should be sufficient in most cases.
+- You can use a `where` clasue to filter the data down.
+- For example:
+
+```sql
+with
+
+requirments as (
+    select * from {{ ref("derived_delius_stg__base_rqmnt") }}
+),
+
+reference_1 as (
+    select * from {{ ref("derived_delius_stg__stg_standard_reference_1") }} 
```

Comment:
all should read as **ref("derived_delius_stg__stg_standard_reference")** without the _1/_2/etc

Reply:
So I took this code, but changed it so that it better represented what we are trying to exhibit in this part of the guidance. None of the models are actually functional models - I just wanted them to have familiar names. To make this clearer I could add disclaimers?

Reply:
ah gotcha, we're just demonstrating importing multiple different sources. makes sense - can leave as is